File: CS/WindowServiceExample/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace WindowServiceExample.ViewModels {
    public class MainViewModel {
        protected MainViewModel() { }

        public static MainViewModel Create() {
            return ViewModelSource.Create(() => new MainViewModel());
        }

        public void ShowFirst() {
            WindowService.Show("FirstWindowView", null, this);
        }
        public void ShowSecond() {
            WindowService.Show("SecondWindowView", null, this);
        }

        public IWindowService WindowService {
            get {
                return this.GetService<IWindowService>();
            }
        }
    }
}

File: VB/WindowServiceExample/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace WindowServiceExample.ViewModels

    Public Class MainViewModel

        Protected Sub New()
        End Sub

        Public Shared Function Create() As MainViewModel
            Return ViewModelSource.Create(Function() New MainViewModel())
        End Function

        Public Sub ShowFirst()
            WindowService.Show("FirstWindowView", Nothing, Me)
        End Sub

        Public Sub ShowSecond()
            WindowService.Show("SecondWindowView", Nothing, Me)
        End Sub

        Public ReadOnly Property WindowService As IWindowService
            Get
                Return GetService(Of IWindowService)()
            End Get
        End Property
    End Class
End Namespace
